users/greatwizard/underglow.c

```diff
@@ -0,0 +1,138 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "underglow.h"
+
+void keyboard_post_init_rgb(void) {
+    user_config_t user_config;
+    user_config.raw = eeconfig_read_user();
+    if (user_config.rgb_layer_change) {
+        rgblight_enable_noeeprom();
+        rgblight_setrgb(RGBLIGHT_COLOR_LAYER_1);
+        rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+    }
+}
+
+bool process_record_rgb(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case RGB_LAYER:
+        if (record->event.pressed) {
+            user_config_t user_config;
+            user_config.raw = eeconfig_read_user();
+            user_config.rgb_layer_change ^= 1;
+            eeconfig_update_user(user_config.raw);
+            if (user_config.rgb_layer_change) {
+                layer_state_set(layer_state);
+            }
+        }
+        return false;
+        break;
+    case RGB_MODE_FORWARD...RGB_MODE_GRADIENT:
+        if (record->event.pressed) {
+            user_config_t user_config;
+            user_config.raw = eeconfig_read_user();
+            if (user_config.rgb_layer_change) {
+                user_config.rgb_layer_change = false;
+                eeconfig_update_user(user_config.raw);
+            }
+        }
+        return true;
+        break;
+    }
+    return true;
+}
+
+layer_state_t layer_state_set_rgb(layer_state_t state) {
+    user_config_t user_config;
+    user_config.raw = eeconfig_read_user();
+    if (!user_config.rgb_layer_change) {
+        return state;
+    }
+    switch (get_highest_layer(state)) {
+        case _QWERTY:
+            #ifdef RGBLIGHT_COLOR_LAYER_1
+                rgblight_setrgb(RGBLIGHT_COLOR_LAYER_1);
+            #endif
+            break;
+        case _WORKMAN:
+            #ifdef RGBLIGHT_COLOR_LAYER_2
+                rgblight_setrgb(RGBLIGHT_COLOR_LAYER_2);
+            #endif
+            break;
+        case _WORKMAN_PROGRAMMER:
+            #ifdef RGBLIGHT_COLOR_LAYER_3
+                rgblight_setrgb(RGBLIGHT_COLOR_LAYER_3);
+            #endif
+            break;
+        case _COLEMAK:
+            #ifdef RGBLIGHT_COLOR_LAYER_4
+                rgblight_setrgb(RGBLIGHT_COLOR_LAYER_4);
+            #endif
+        case _DVORAK:
+            #ifdef RGBLIGHT_COLOR_LAYER_5
+                rgblight_setrgb(RGBLIGHT_COLOR_LAYER_5);
+            #endif
```

Comment:
Just a heads up, these may not work as indended, as the default layer is 0, and you'd need to check `default_layer_state`

Reply:
Thanks I correct it after testing everything :)